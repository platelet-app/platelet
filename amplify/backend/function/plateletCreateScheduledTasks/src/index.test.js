const indexModule = require("./index");

const mockScheduledTasks = [
    {
        id: "taskId",
        tenantId: "tenantId",
        priority: "MEDIUM",
        requesterContact: {
            name: "name",
            telephoneNumber: "phone",
        },
        establishmentLocation: {
            id: "establishmentLocationId",
            listed: 1,
            _version: 1,
        },
        pickUpLocation: {
            id: "pickUpLocationId",
            listed: 1,
            _version: 1,
        },
        dropOffLocation: {
            id: "dropOffLocationId",
            contact: {
                name: "name",
                telephoneNumber: "phone",
            },
            line1: "line1",
            line2: "line2",
            line3: "line3",
            town: "town",
            county: "county",
            country: "country",
            postcode: "postcode",
            what3words: "what3words",
            tenantId: "tenantId",
            listed: 0,
            _version: 1,
        },
        deliverables: {
            items: [
                {
                    id: "deliverableId",
                    deliverableType: { id: "deliverableType", label: "bloods" },
                    deliverableTypeDeliverablesId:
                        "deliverableTypeDeliverablesId",
                    count: 1,
                    unit: "unit",
                },
            ],
        },
        _version: 1,
        _deleted: null,
        _lastChangedAt: 1620000000000,
    },
];

jest.mock(
    "/opt/appSyncRequest",
    () => {
        return {
            request: jest.fn(),
        };
    },
    { virtual: true }
);
jest.mock(
    "/opt/graphql/mutations",
    () => {
        return {
            createTask: "createTaskQuery",
            createLocation: "createLocationQuery",
            createDeliverable: "createDeliverableQuery",
        };
    },
    { virtual: true }
);

const appsyncModule = require("/opt/appSyncRequest");

describe("plateletCreateScheduledTasks", () => {
    afterEach(() => jest.restoreAllMocks());
    beforeAll(() => {
        jest.useFakeTimers("modern");
        jest.setSystemTime(new Date("2021-05-01T00:00:00.000Z"));
    });
    test("create some scheduled tasks", async () => {
        appsyncModule.request
            .mockResolvedValueOnce({
                json: () => ({
                    data: {
                        listScheduledTasks: {
                            items: mockScheduledTasks,
                            nextToken: null,
                        },
                    },
                }),
            })
            .mockResolvedValueOnce({
                json: () => ({
                    data: { createLocation: { id: "newLocationId" } },
                }),
            })
            .mockResolvedValueOnce({
                json: () => ({
                    data: { createTask: { id: "newTaskId" } },
                }),
            })
            .mockResolvedValueOnce({
                json: () => ({
                    data: { createDeliverable: { id: "newDeliverableId" } },
                }),
            });

        await indexModule.handler();
        expect(appsyncModule.request).toMatchSnapshot();
    });
});
