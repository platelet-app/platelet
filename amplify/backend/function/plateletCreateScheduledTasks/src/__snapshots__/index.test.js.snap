// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plateletCreateScheduledTasks create some scheduled tasks 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "query": "
    query ListScheduledTasks(
        $filter: ModelScheduledTaskFilterInput
        $limit: Int
        $nextToken: String
    ) {
        listScheduledTasks(
            filter: $filter
            limit: $limit
            nextToken: $nextToken
        ) {
            items {
                id
                tenantId
                cronExpression
                pickUpLocation {
                    id
                    listed
                    tenantId
                    name
                    contact {
                        name
                        telephoneNumber
                        mobileNumber
                        emailAddress
                        ward
                        line1
                        line2
                        line3
                        town
                        county
                        state
                        country
                        postcode
                        what3words
                    }
                    ward
                    line1
                    line2
                    line3
                    town
                    county
                    state
                    country
                    postcode
                    what3words
                }
                dropOffLocation {
                    id
                    listed
                    tenantId
                    name
                    contact {
                        name
                        telephoneNumber
                        mobileNumber
                        emailAddress
                        ward
                        line1
                        line2
                        line3
                        town
                        county
                        state
                        country
                        postcode
                        what3words
                    }
                    ward
                    line1
                    line2
                    line3
                    town
                    county
                    state
                    country
                    postcode
                    what3words
                }
                establishmentLocation {
                    id
                    listed
                    name
                    tenantId
                }
                deliverables {
                    items {
                        count
                        deliverableTypeDeliverablesId
                        unit
                    }
                }
                requesterContact {
                    name
                    telephoneNumber
                    mobileNumber
                    emailAddress
                    ward
                    line1
                    line2
                    line3
                    town
                    county
                    state
                    country
                    postcode
                    what3words
                }
                priority
                disabled
            }
            nextToken
            startedAt
        }
    }
",
        "variables": {
          "nextToken": null,
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "createLocationQuery",
        "variables": {
          "input": {
            "archived": 0,
            "contact": {
              "name": "name",
              "telephoneNumber": "phone",
            },
            "country": "country",
            "county": "county",
            "line1": "line1",
            "line2": "line2",
            "line3": "line3",
            "listed": 0,
            "name": undefined,
            "postcode": "postcode",
            "state": undefined,
            "tenantId": "tenantId",
            "town": "town",
            "ward": undefined,
            "what3words": "what3words",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "createTaskQuery",
        "variables": {
          "input": {
            "archived": 0,
            "dateCreated": "2021-05-01",
            "dropOffLocationId": "newLocationId",
            "establishmentLocationId": "establishmentLocationId",
            "pickUpLocationId": "pickUpLocationId",
            "priority": "MEDIUM",
            "requesterContact": {
              "name": "name",
              "telephoneNumber": "phone",
            },
            "status": "PENDING",
            "tenantId": "tenantId",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
    [
      {
        "query": "createDeliverableQuery",
        "variables": {
          "input": {
            "archived": 0,
            "count": 1,
            "deliverableTypeDeliverablesId": "deliverableTypeDeliverablesId",
            "taskDeliverablesId": "newTaskId",
            "tenantId": "tenantId",
            "unit": "unit",
          },
        },
      },
      "https://api.example.com/graphql",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
    {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
